File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/LnurlWithdraw.kt
package com.example.kotlinmpplib

import breez_sdk.*
class LnurlWithdraw {
    fun lnurl_withdraw(sdk: BlockingBreezServices) {
        // ANCHOR: lnurl-withdraw
        // Endpoint can also be of the form:
        // lnurlw://domain.com/lnurl-withdraw?key=val
        val lnurlWithdrawUrl = "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7..."
        try {
            val inputType = parseInput(lnurlWithdrawUrl)
            if (inputType is InputType.LnUrlWithdraw) {
                val requestData = inputType.data
                val amountMsat = requestData.minWithdrawable
                val comment = "Any comment"
                sdk.withdrawLnurl(LnUrlWithdrawRequest(requestData, amountMsat, comment))
            }
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: lnurl-withdraw
    }
}

File: snippets/swift/BreezSDKExamples/Sources/LnurlWithdraw.swift
//
//  LnurlWithdraw.swift
//
//
//  Created by ruben on 14/11/2023.
//

import Foundation
import BreezSDK

func withdraw(sdk: BlockingBreezServices) -> LnUrlWithdrawResult? {
    // ANCHOR: lnurl-withdraw
    // Endpoint can also be of the form:
    // lnurlw://domain.com/lnurl-withdraw?key=val
    var response: LnUrlWithdrawResult?
    let lnurlWithdrawUrl = "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4exctthd96xserjv9mn7um9wdekjmmw843xxwpexdnxzen9vgunsvfexq6rvdecx93rgdmyxcuxverrvcursenpxvukzv3c8qunsdecx33nzwpnvg6ryc3hv93nzvecxgcxgwp3h33lxk"

    if let inputType = try? parseInput(s: lnurlWithdrawUrl){
        if case.lnUrlWithdraw(let `data`) = inputType {
            let amountMsat = data.maxWithdrawable
            let description = "Test withdraw"
            response = try? sdk.withdrawLnurl(
                request: LnUrlWithdrawRequest(
                    data: data,
                    amountMsat: amountMsat,
                    description: description
                )
            )
        }
    }
    // ANCHOR_END: lnurl-withdraw
    return response
}
